visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts

```diff
@@ -62,6 +62,10 @@ export class EdgeMetricDataService implements StoreSubscriber, BlacklistSubscrib
 			return keys
 		}
 
+		if (amountOfEdgePreviews === 0) {
```

Comment:
I would place this condition at the top of the method. If I understand it correctly, we don not need to retrieve the `nodeEdgeMetrics` in this case.
Furthermore, I would just return an empty array like in the case the `nodeEdgeMetrics` are `undefined`. Thus, no further changes would be neccessary and we can revert the change here codeMap.actions.service.ts:63.